src/cat-data-service/src/service/v1/registration.rs

```diff
@@ -0,0 +1,348 @@
+use crate::{
+    service::{handle_result, Error},
+    state::State,
+};
+use axum::{extract::Path, routing::get, Router};
+use event_db::types::snapshot::{Delegator, EventId, Voter};
+use std::sync::Arc;
+
+pub fn registration(state: Arc<State>) -> Router {
+    Router::new()
+        .route(
+            "/registration/voter/:voting_key",
+            get({
+                let state = state.clone();
+                move |path| async {
+                    handle_result(voter_by_latest_event_exec(path, state).await).await
+                }
+            }),
+        )
+        .route(
+            "/registration/voter/:voting_key/:event",
+            get({
+                let state = state.clone();
+                move |path| async { handle_result(voter_by_event_exec(path, state).await).await }
+            }),
+        )
+        .route(
+            "/registration/delegations/:stake_public_key",
+            get({
+                let state = state.clone();
+                move |path| async {
+                    handle_result(delegations_by_latest_event_exec(path, state).await).await
+                }
+            }),
+        )
+        .route(
+            "/registration/delegations/:stake_public_key/:event",
```

Comment:
Same comment as above.
`:event` isn't a path parameter `eid` is an optional query parameter in the format `?eid=77`.